.github/workflows/nightly-build-publish.yml

```diff
@@ -14,34 +14,124 @@ on:
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: ./build/dockerfiles/Dockerfile
+          platforms: linux/${{ matrix.arch }}
+          tags:  ${{ env.IMAGE }}:${{ env.VERSION }}-${{ matrix.arch }}
+          push: true 
+      -
+        id: result
+        name: "Build result outputs version"
+        if: ${{ success() }}
+        run: echo "::set-output name=${{ matrix.arch }}::${{ env.VERSION }}-${{ matrix.arch }}"
+
+  create-manifest:
+    if: always()
+    needs: nightly-build-publish
+    runs-on: ubuntu-latest
+    steps:
+      -
+        name: "Checkout source code"
+        uses: actions/checkout@v1
+        with:
+          fetch-depth: 0
       - 
         name: Prepare
         id: prep
         run: |
           SHORT_SHA1=$(git rev-parse --short HEAD)
           echo ::set-output name=short_sha1::${SHORT_SHA1}
-          # VERSION file contains 7.yy.0-SNAPSHOT but for nightlies we want just a nightly tag
-          # VERSION=$(head -n 1 VERSION)
-          # echo ::set-output name=version::${VERSION}
-          echo ::set-output name=version::nightly
-          IMAGE=che-devfile-registry
-          echo ::set-output name=image::${IMAGE}
-          PLATFORMS=$(cat PLATFORMS)
-          echo ::set-output name=platforms::${PLATFORMS}
+      -
+        name: "Docker quay.io Login"
+        uses: docker/login-action@v1
+        with:
+          registry: quay.io
+          username: ${{ secrets.QUAY_USERNAME }}
+          password: ${{ secrets.QUAY_PASSWORD }}
+      -  
+        name: "Create and push manifest"
+        run: |
+          AMEND=""
+          AMD64_VERSION="${{ needs['nightly-build-publish'].outputs.amd64 }}"
+          if [ -n "$AMD64_VERSION" ]; then
+            AMEND+=" --amend ${{ env.IMAGE }}:$AMD64_VERSION";
+          fi
+          ARM64_VERSION="${{ needs['nightly-build-publish'].outputs.arm64 }}"
+          if [ -n "$ARM64_VERSION" ]; then
+            AMEND+=" --amend ${{ env.IMAGE }}:$ARM64_VERSION";
+          fi
+          PPC64LE_VERSION="${{ needs['nightly-build-publish'].outputs.ppc64le }}"
+          if [ -n "$PPC64LE_VERSION" ]; then
+            AMEND+=" --amend ${{ env.IMAGE }}:$PPC64LE_VERSION";
+          fi
+          S390X_VERSION="${{ needs['nightly-build-publish'].outputs.s390x }}"
+          if [ -n "$S390X_VERSION" ]; then
+            AMEND+=" --amend ${{ env.IMAGE }}:$S390X_VERSION";
+          fi
+          if [ -z "$AMEND" ]; then
+            echo "[!] The job 'nightly-build-publish' didn't provide any outputs. Can't create the manifest list."
+            exit 1;
+          fi
+          docker manifest create ${{ env.IMAGE }}:${{ env.VERSION }} $AMEND
+          docker manifest push ${{ env.IMAGE }}:${{ env.VERSION }}
+          docker manifest create ${{ env.IMAGE }}:${{ steps.prep.outputs.short_sha1 }} $AMEND
+          docker manifest push ${{ env.IMAGE }}:${{ steps.prep.outputs.short_sha1 }}
+
+  build-images:
+    if: always()
```

Comment:
Do we need this line if it's always going to happen?

Reply:
In this case, it will always run after nightly-build-publish job has completed regardless of whether it is successful. Also just to confirm, should this be executed only when nightly-build-publish is successful or it can be executed as a separate standalone job?

Reply:
If `nightly-build-publish` fails then building an image for that job wouldn't really serve much purpose, right? Am I understanding it correctly?

Reply:
Yes, but build-images is building base images and happy path image. This should be executed only when nightly-build-publish(which builds che-devfile-registry image) job is successful? 

Also, to add more on this. Should we keep nightly build workflow similar to release workflow? wherein if build fails on 1 arch then further steps should not be executed.

Reply:
cc @nickboldt how do we handle this in other places? My instinct would say we should fail on any arch failure.

Reply:
In CRW we *used* to have the build workflow scenario where if a multiarch build failed on one arch, we'd quietly remove those arches from the container.yaml used for Brew and just build it for the passing arches. This led to more problems than it was worth...
* PRO: could continue the overall build stack / release even when Z or P was broken
* PRO: if missing upstream artifacts, wouldn't waste time/resources/data centre heat building something we know was going to fail (or which used older artifacts and would therefore lead to unexpected containter content [eg., theia container 2.8 built from theia tarballs from 2.7]; this is NOT a use case for Che since Che doesn't do previously-bootstrapped offline builds, and will always fetch latest from GH and npm and go registries.

* CON: Z Dev/QE could be lost at sea wondering where their updated containers were, and why only x and P got updated in quay
* CON: might end up with a case where we can't deploy on Z or P because the operator-metadata payload was incomplete (eg., if theia or traefik failed to build)

Do we run automated tests on all arches (on push or for PRs)? can those tests be skipped if the arch-specific containers failed to build? Or do we want to gate a given push/PR from being released if an arch is missing? 

I'd say the answer may depend on flakiness of the arch-specific builds. If z fails to build 1 time in 50, then its absence should be a bug that would prevent the release. Buf it if fails 1 time in 3, then maybe it's worth skipping over it if it's going to hinder progress 66.6% of the time.